app/src/main/java/org/desarrolladorslp/teckersapp/MainActivity.kt

```diff
@@ -105,25 +111,43 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
             .addOnCompleteListener(this) { task ->
                 if (task.isSuccessful) {
                     val user = auth.currentUser
-                    setProfileInformation(acct)
-                    updateUI(user)
+                    val firebaseTokenId = user?.zze()?.zzd()
+
+                    setProfileInformation(acct, firebaseTokenId)
                 } else {
-                    Snackbar.make(main_layout, "Authentication Failed.", Snackbar.LENGTH_SHORT).show()
+                    Snackbar.make(main_layout, "Authentication Failed.", Snackbar.LENGTH_SHORT)
+                        .show()
                     updateUI(null)
                 }
 
             }
     }
 
-    private fun setProfileInformation(acct: GoogleSignInAccount?){
+    private fun setProfileInformation(acct: GoogleSignInAccount?, firebaseTokenId: String?) {
 
         if (acct != null) {
-            user = User(acct)
+
+            var loginService = RetrofitManager.instance()?.create(LoginService::class.java);
+
+            var loginCall = loginService?.login("firebase", firebaseTokenId)
+
+            loginCall?.enqueue(object : Callback<LoggedUser> {
+                override fun onResponse(call: Call<LoggedUser>, response: Response<LoggedUser>) {
+                    if (response.code() == 200) {
```

Comment:
If `response.code()` is not 200 an error message should be shown to the user... maybe those scenarios can be handled from an interceptor so only 200 responses arrives to this method